crates/ratchet-core/src/ops/softmax.rs

```diff
@@ -13,6 +13,27 @@ pub struct Softmax {
     dim: usize,
 }
 
+//type templated to generate all the variants.
+//functions are generated for each variant
+//the generated functions are then used to generate the kernel
+//pub<TypeLevelGynmastics> define() -> Kernel(String)
+//Have a nice crate that generates the kernel with Comptime & Runtime
+// CLI and Runtime can be different.
+//enum Assert<const b: bool> {}
+//trait IsTrue {}
+//trait IsFalse {}
+//
+//impl IsTrue for Assert<true> {}
+//impl IsFalse for Assert<false> {}
+//
+//type Scalar = 0;
+//type Vec2 = 2;
+//type Vec4 = 4;
```

Comment:
I was thinking something like
`type Vec2<T> = AccessGranularity<T, 2>`

Reply:
This is nice, let me get back to you.